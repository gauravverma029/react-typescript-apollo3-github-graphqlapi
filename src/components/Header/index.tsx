import React from "react";

const Header: React.FC = () => {
  return (
    <div className="w-screen bg-gray-900">
      <div className="p-2 flex justify-center">
        <div className="text-2xl font-semibold text-white">Github</div>
      </div>
    </div>
  );
};
export default Header;
