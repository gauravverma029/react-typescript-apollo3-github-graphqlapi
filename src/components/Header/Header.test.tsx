import React from "react";
import { cleanup, render } from "@testing-library/react";

import Header from "./index";

afterEach(cleanup);

describe("Header Component", () => {
  test("Should Render", async () => {
    const { asFragment } = render(<Header />);
    const html = asFragment();
    expect(html).toMatchSnapshot();
  });

  test("Should Contain Text 'Github'", async () => {
    const { getByText } = render(<Header />);
    getByText("Github");
  });

  test("Should Contain div tag for github text", async () => {
    const { getByText } = render(<Header />);
    expect(getByText("Github").tagName).toBe("DIV");
  });

  test("Should Contain ClassName", async () => {
    const { getByText } = render(<Header />);
    expect(getByText("Github").className).toBe(
      "text-2xl font-semibold text-white"
    );
  });
});
