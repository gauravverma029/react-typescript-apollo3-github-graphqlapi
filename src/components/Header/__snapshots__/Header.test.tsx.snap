// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Should Render 1`] = `
<DocumentFragment>
  <div
    class="w-screen bg-gray-900"
  >
    <div
      class="p-2 flex justify-center"
    >
      <div
        class="text-2xl font-semibold text-white"
      >
        Github
      </div>
    </div>
  </div>
</DocumentFragment>
`;
